### ISSUE TITLE
UnassignedVariableUsage QuickFix - Ignore Once puts annotation in the wrong place

### ISSUE THREAD

**Issue by daFreeMan** [rubberduck-vba/Rubberduck#3283]
Starting with this:

```VBA
Public Sub ArchiveAllSourceDataByType(ByVal filesToProcess As String, ByVal clinicList As ListBox)
'TODO Add Error Handling

  LogManager.Log TraceLevel, "ArchiveDataProcess.ArchiveAllSourceDataByType"
  
  'get the file extension for the type of data we're archiving
  Dim FileExtension As String
  FileExtension = Tools.GetGeneralConfig(CallLocation, FileExtensionCategory, filesToProcess)
  
  Dim ArchiveData As ZipToolsData
  
  Dim Archiver As ZipTools
  Dim Index As Long
  Dim SQLString As String
  Dim ExcelRS As ADODB.Recordset
  For Index = 0 To clinicList.ListCount - 1
    'update the screen so the user knows what's going on
    clinicList.Selected(Index) = True
    'find out what files this clinic should have
    SQLString = "SELECT ExcelFileName " & _
                "  FROM " & ExcelConfigTable & _
                " WHERE ClinicID = " & clinicList.Column(0, Index) & _
                "   AND ExcelFileType = '" & filesToProcess & "'"
    Set ExcelRS = New ADODB.Recordset
    ExcelRS.Open Source:=SQLString, ActiveConnection:=CStr(TempVars!ConString)
    While Not ExcelRS.EOF And Not ExcelRS.BOF
      ArchiveASingleFile ArchiveData, Archiver, filesToProcess, ExcelRS!ExcelFileName & FileExtension
      ExcelRS.MoveNext
    Wend
    clinicList.Selected(Index) = False
    ExcelRS.Close
  Next
  
'CleanExit:
  If ExcelRS.State <> adStateClosed Then
    ExcelRS.Close
  End If
  
  Exit Sub
  
End Sub
```
![rd quickfix ignore once before](https://user-images.githubusercontent.com/11889733/29537608-5766200e-8690-11e7-9a75-e6641a68848f.PNG)

The `UnassignedVariableUsage` Ignore Once option puts the annotation in the wrong place:

```VBA
Public Sub ArchiveAllSourceDataByType(ByVal filesToProcess As String, ByVal clinicList As ListBox)
'TODO Add Error Handling

  LogManager.Log TraceLevel, "ArchiveDataProcess.ArchiveAllSourceDataByType"
  
  'get the file extension for the type of data we're archiving
  Dim FileExtension As String
  FileExtension = Tools.GetGeneralConfig(CallLocation, FileExtensionCategory, filesToProcess)
  
  Dim ArchiveData As ZipToolsData
  
  Dim Archiver As ZipTools
  Dim Index As Long
  Dim SQLString As String
  Dim ExcelRS As ADODB.Recordset
  For Index = 0 To clinicList.ListCount - 1
    'update the screen so the user knows what's going on
    clinicList.Selected(Index) = True
'@Ignore UnassignedVariableUsage
    'find out what files this clinic should have
    SQLString = "SELECT ExcelFileName " & _
                "  FROM " & ExcelConfigTable & _
                " WHERE ClinicID = " & clinicList.Column(0, Index) & _
                "   AND ExcelFileType = '" & filesToProcess & "'"
    Set ExcelRS = New ADODB.Recordset
    ExcelRS.Open Source:=SQLString, ActiveConnection:=CStr(TempVars!ConString)
    While Not ExcelRS.EOF And Not ExcelRS.BOF
      ArchiveASingleFile ArchiveData, Archiver, filesToProcess, ExcelRS!ExcelFileName & FileExtension
      ExcelRS.MoveNext
    Wend
    clinicList.Selected(Index) = False
    ExcelRS.Close
  Next
  
'CleanExit:
  If ExcelRS.State <> adStateClosed Then
    ExcelRS.Close
  End If
  
  Exit Sub
  
End Sub
```
![rd quickfix ignore once after](https://user-images.githubusercontent.com/11889733/29537616-5d74ae98-8690-11e7-8952-05372a1e18d3.PNG)

**Comment by daFreeMan** [comment]
Oddly, even after moving the annotation, it still doesn't seem to ignore this issue:

![rd quickfix ignore once still not ignored](https://user-images.githubusercontent.com/11889733/29537793-0c6c2d7c-8691-11e7-9f83-d7f91f6dd108.PNG)

**Comment by comintern** [comment]
This was fixed by changes in 3a5e485.
